npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -55,21 +55,22 @@ def manifest_dependencies
         dependency_set = DependencySet.new
 
         package_files.each do |file|
+          json = JSON.parse(file.content)
           # TODO: Currently, Dependabot can't handle flat dependency files
           # (and will error at the FileUpdater stage, because the
           # UpdateChecker doesn't take account of flat resolution).
-          next if JSON.parse(file.content)["flat"]
+          next if json["flat"]
 
           DEPENDENCY_TYPES.each do |type|
-            deps = JSON.parse(file.content)[type] || {}
-            deps.each do |name, requirement|
+            (json[type] || {}).each do |name, requirement|
               next unless requirement.is_a?(String)
 
-              requirement = "*" if requirement == ""
-              dep = build_dependency(
-                file: file, type: type, name: name, requirement: requirement
+              dependency_set << build_dependency(
+                file: file,
+                type: type,
+                name: name,
+                requirement: requirement.empty? ? "*" : requirement
               )
-              dependency_set << dep if dep
```

Comment:
Related to the change to the dependency_set, thoughts on keeping this behaviour here so we don't need to change the common class?